llvm-spirv/lib/SPIRV/OCLUtil.cpp

```diff
@@ -531,6 +531,24 @@ class OCLBuiltinFuncMangleInfo : public SPIRV::BuiltinFuncMangleInfo {
       addVoidPtrArg(1);
       addUnsignedArg(2);
       addUnsignedArg(3);
+      // TODO: REMOVE BEFORE COMMIT!!!
+      // Here and after we are translating calls of blocking pipe SPIR-V
+      // instructions to calls to some OpenCL-like built-ins, so LLVM IR before
+      // and after translation changes (actually same is happening for
+      // non-blocking pipes, which is considered as a bug that must be fixed).
+      // Instead of it we can not create special rules for ReadPipeBlockingINTEL
+      // and WritePipeBlockingINTEL, but emit SPIR-V looking like LLVM IR.
+      // Pros: No nonexistent OpenCL built-ins used.
+      // Cons: It will add mangling with translation back from SPIR-V code to
+      // LLVM IR.
+    } else if (UnmangledName == "read_pipe_bl_2" ||
```

Comment:
I'm in favor of keeping current implementation as is:
  1. It is expected (or highly desired) that blocking and non-blocking builtins look similar in IR (both before SPIRV and after). It would be really weird if one of them is `__read_pipe_2` and the other is `_Z24__spirv_ReadPipeBlockingU8ocl_pipeSomething`.
  2. In current implementation of SPIR-V friendly IR, pipe builtins are mangled with user type, e.g. a read from a `pipe float` and a read from a `pipe int` result it two different functions. This makes middle-end job unnecessary difficult.

Reply:
I was thinking of making LLVM IR output from the translator different depending on option passed to it, like "llvm-spirv -r -opencl ..." will result in generating OpenCL-like IR and default "llvm-spirv -r ..." will result in SPIR-V like IR.